### PR TITLE
Fixes #26040: Have a stable webapp log file - packaging part

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/26040

### CODE REVIEW

File: rudder-server/SOURCES/Makefile [hunk 1]
```diff
@@ -161,7 +161,8 @@ install: build rudder-server-version initial-promises initial-ncf
 	mkdir -p $(DESTDIR)/var/rudder/run
 	mkdir -p $(DESTDIR)/var/rudder/ldap/openldap-data
 	mkdir -p $(DESTDIR)/var/log/rudder/apache2/
-	mkdir -p $(DESTDIR)/var/log/rudder/webapp
+	mkdir -p $(DESTDIR)/var/log/rudder/webapp/jetty
+	mkdir -p $(DESTDIR)/var/log/rudder/webapp/jvm
```
Comment (fanf) [inline rudder-server/SOURCES/Makefile:165]: perhpas also `mkdir -p $(DESTDIR)/var/log/rudder/webapp/api` depending of what we decide for that directory

**Review by fanf — COMMENTED** [review]
Waiting for the decision about API directory, but else lgtm

File: rudder-server/SOURCES/rudder-jetty.conf [hunk 2]
```diff
@@ -54,7 +54,7 @@ JAVA_GC=${JAVA_GC:="-XX:+UseG1GC
 
 # GC log: file path, log level (jdk format, see:
```
Comment (fanf) [inline rudder-server/SOURCES/rudder-jetty.conf:55]: I think we need to add the production mode here

File: rudder-server/SOURCES/Makefile (hunk 1 above)
Comment (fanf) [reply rudder-server/SOURCES/Makefile:165]: Since we added webapp/api, the additional mkdir is needed

**Review by fanf — CHANGES_REQUESTED** [review]
Some changes needed because of what is done in https://github.com/Normation/rudder/pull/6076

File: rudder-server/SOURCES/rudder-jetty.conf (hunk 2 above)
Comment (fanf) [reply rudder-server/SOURCES/rudder-jetty.conf:55]: ok, it's already therer, I just don't know where it is defined: 
```
root@rudder-snapshot:~# ps aux | grep rudder | grep prod
root       11235  1.1 72.9 3414860 1470352 ?     Sl   00:54   6:56 /usr/lib/jvm/java-17-openjdk-amd64/bin/java -Djava.io.tmpdir=/tmp -Djetty.home=/opt/rudder/jetty -Djetty.base=/opt/rudder/etc/rudder-jetty-base -server -Xms1024m -Xmx1024m -XX:+UseG1GC -XX:+UseStringDeduplication -XX:+AlwaysPreTouch -Xlog:gc=info,gc+cpu=info,gc+stringdedup=info:file=/var/log/rudder/webapp/jvm-gc.log:utctime,pid,level,tags:filecount=5,filesize=50M -XX:+CrashOnOutOfMemoryError -Dfile.encoding=UTF-8 
-Drun.mode=production  // <- HERE
-Drudder.configFile=/opt/rudder/etc/rudder-web.properties -Drudder.authFile=/opt/rudder/etc/rudder-users.xml -Dlogback.configurationFile=/opt/rudder/etc/logback.xml --class-path /opt/rudder/etc/rudder-jetty-base/resources:/opt/rudder/jetty/lib/logging/slf4j-api-2.0.5.jar:/opt/rudder/jetty/lib/logging/jetty-slf4j-impl-10.0.18.jar:/opt/rudder/jetty/lib/jetty-servlet-api-4.0.6.jar:/opt/rudder/jetty/lib/jetty-http-10.0.18.jar:/opt/rudder/jetty/lib/jetty-server-10.0.18.jar:/opt/rudder/jetty/lib/jetty-xml-10.0.18.jar:/opt/rudder/jetty/lib/jetty-util-10.0.18.jar:/opt/rudder/jetty/lib/jetty-io-10.0.18.jar:/opt/rudder/jetty/lib/jetty-security-10.0.18.jar:/opt/rudder/jetty/lib/jetty-servlet-10.0.18.jar:/opt/rudder/jetty/lib/jetty-webapp-10.0.18.jar:/opt/rudder/jetty/lib/jetty-deploy-10.0.18.jar:/opt/rudder/jetty/lib/setuid/jetty-setuid-java-1.0.4.jar org.eclipse.jetty.xml.XmlConfiguration file.encoding=UTF-8 java.version=17.0.9 jetty.base=/opt/rudder/etc/rudder-jetty-base jetty.base.uri=file:///opt/rudder/etc/rudder-jetty-base jetty.deploy.monitoredDir=/opt/rudder/share/webapps jetty.deploy.scanInterval=0 jetty.home=/opt/rudder/jetty jetty.home.uri=file:///opt/rudder/jetty jetty.http.host=127.0.0.1 jetty.http.port=8080 jetty.httpConfig.requestHeaderSize=8192 jetty.httpConfig.sendDateHeader=false jetty.httpConfig.sendServerVersion=false jetty.pid=/var/rudder/run/rudder-jetty.pid jetty.setuid.groupName=root jetty.setuid.startServerAsPrivileged=true jetty.setuid.umask=0007 jetty.setuid.userName=root jetty.state=/opt/rudder/etc/rudder-jetty-base/rudder-jetty.state jetty.webapp.addServerClasses=org.eclipse.jetty.logging.,file:///opt/rudder/jetty/lib/logging/,org.slf4j. logback.configurationFile=/opt/rudder/etc/logback.xml OPTIONS=Server rudder.authFile=/opt/rudder/etc/rudder-users.xml rudder.configFile=/opt/rudder/etc/rudder-web.properties run.mode=production runtime.feature.alpn=true slf4j.version=2.0.5 /opt/rudder/jetty/etc/jetty-bytebufferpool.xml /opt/rudder/etc/rudder-jetty-base/etc/console-capture.xml /opt/rudder/jetty/etc/jetty-pid.xml /opt/rudder/jetty/etc/jetty-threadpool.xml /opt/rudder/jetty/etc/jetty.xml /opt/rudder/jetty/etc/jetty-state.xml /opt/rudder/jetty/etc/jetty-webapp.xml /opt/rudder/jetty/etc/jetty-deploy.xml /opt/rudder/jetty/etc/jetty-http.xml /opt/rudder/jetty/etc/jetty-setuid.xml

```

**Comment by VinceMacBuche** [comment]
Commit modified

**Review by fanf — APPROVED** [review]
LGTM

**Pull request merged**